libcrux-kem/src/kem.rs

```diff
@@ -576,54 +474,101 @@ impl PublicKey {
                 ))
             }
 
-            PublicKey::XWingKemDraft02(XWingKemDraft02PublicKey { pk_m, pk_x }) => {
+            PublicKey::XWingKemDraft06(XWingKemDraft06PublicKey { pk_m, pk_x }) => {
                 let seed = mlkem_rand(rng)?;
                 let (ct_m, ss_m) = mlkem768::encapsulate(pk_m, seed);
                 let (ek_x, ct_x) = libcrux_ecdh::x25519_key_gen(rng)?;
                 let ss_x = x25519_derive(pk_x, &ek_x)?;
 
                 Ok((
-                    Ss::XWingKemDraft02(
-                        ss_m,
-                        ss_x,
-                        X25519PublicKey(ct_x.0.clone()),
-                        X25519PublicKey(pk_x.0.clone()),
-                    ),
-                    Ct::XWingKemDraft02(ct_m, X25519PublicKey(ct_x.0.clone())),
+                    Ss::XWingKemDraft06(xwing::combiner(
+                        &ss_m,
+                        ss_x.as_ref(),
+                        ct_x.as_ref(),
+                        pk_x.as_ref(),
+                    )),
+                    Ct::XWingKemDraft06(ct_m, X25519PublicKey(ct_x.0)),
                 ))
             }
+        }
+    }
+
+    /// Encapsulate a shared secret to the provided `pk` and return the `(Key, Enc)` tuple.
+    pub fn encapsulate_derand(&self, seed: &[u8]) -> Result<(Ss, Ct), Error> {
+        match self {
+            PublicKey::X25519(pk) => {
+                let new_sk = X25519PrivateKey::try_from(seed)?; // clamps
+                let new_pk =
+                    libcrux_ecdh::secret_to_public(libcrux_ecdh::Algorithm::X25519, &new_sk)?;
```

Comment:
```suggestion
                    x25519_secret_to_public(&new_sk)?;
```

AFAICT this also means we don't need the `impl TryFrom<Vec<u8>>` for X25519 public keys.